src/senaite/jsonapi/catalog.py

```diff
@@ -64,12 +65,22 @@ def __call__(self, query):
     def __repr__(self):
         return "<Catalog %s>" % self.get_catalog().getId()
 
-    def get_catalog(self, default="portal_catalog"):
+    def get_catalog(self, default="uid_catalog"):
         name = req.get("catalog")
         if not name:
-            # get the mapped catalog for the given portal_type
-            catalogs = senaiteapi.get_catalogs_for(self.portal_type)
-            name = catalogs[0].getId() if len(catalogs) > 0 else default
+            catalogs = []
+            for portal_type in to_list(self.portal_type):
+                # get the mapped catalog for the given portal_type
+                mapped_catalogs = senaiteapi.get_catalogs_for(portal_type)
+                if len(mapped_catalogs) > 0:
+                    catalogs.append(mapped_catalogs[0])
```

Comment:
I decided here that always the first mapped catalog is the primary and check then below with the `set` if this primary is common or if we have two different primary catalogs.

But besides this, I think we have a design issue here as well in the form that the `portal_type` is expected as mandatory for the `search` route and I guess this should not be.

Reply:
> But besides this, I think we have a design issue here as well in the form that the `portal_type` is expected as mandatory for the `search` route and I guess this should not be.

I agree. I think this is why `portal_catalog` existed in first place. Problem is keeping this catalog with all objects indexed is too expensive for maybe the edge case of willing to search for something regardless of the type. For what I see, `portal_catalog` in this case is only used to retrieve the catalog, so maybe we can skip the retrieval of catalogs for `portal_type` if the param `catalog` is passed through the request. Otherwise, if neither `portal_catalog` nor `catalog` params are present, simply fallback to `uid_catalog`?

Reply:
We skip already the retrieval of the catalogs if the catalog is named as a parameter:
```python
    def get_catalog(self, default="uid_catalog"):
        name = req.get("catalog")
        if not name:
            catalogs = []
            for portal_type in to_list(self.portal_type):
                # Get the mapped catalog for the given portal_type
                mapped_catalogs = senaiteapi.get_catalogs_for(
                    portal_type, default=default)
                # NOTE: We consider the first mapped catalog as the primary!
                if len(mapped_catalogs) > 0:
                    catalogs.append(mapped_catalogs[0])

            # multi catalog query detected -> fall back to default
            if len(set(catalogs)) > 1:
                name = default
            else:
                name = catalogs[0].getId() if len(catalogs) > 0 else default

        return senaiteapi.get_tool(name, default=default)
```

Note that the `to_list` will return and empty list `[]` in case no `portal_type` is passed, also see here: https://github.com/senaite/senaite.jsonapi/blob/2.x/src/senaite/jsonapi/underscore.py#L151-L185